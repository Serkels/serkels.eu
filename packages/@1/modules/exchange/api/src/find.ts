//

import { Exchange_Filter } from "@1.modules/exchange.domain";
import {
  next_auth_procedure,
  procedure,
  router,
  type inferProcedureInput,
} from "@1.modules/trpc";
import type { Prisma } from "@prisma/client";
import { endOfYesterday } from "date-fns";
import { match, P } from "ts-pattern";
import { z } from "zod";

//

const public_schema = z.object({
  category: z.string().optional(),
  cursor: z.string().optional(),
  limit: z.number().min(1).max(10).default(10),
  search: z.string().optional(),
});

const public_procedure = procedure.input(public_schema);
const private_procedure = next_auth_procedure.input(
  public_schema.extend({
    filter: Exchange_Filter.default(Exchange_Filter.Enum.ALL),
  }),
);

export default router({
  public: public_procedure.query(find_resolver),
  private: private_procedure.query(find_resolver),
});

async function find_resolver(
  options:
    | inferProcedureInput<typeof public_procedure>
    | inferProcedureInput<typeof private_procedure>,
) {
  type ExchangeWhere = Prisma.ExchangeWhereInput;
  const {
    ctx: { prisma },
    input: { limit, category, cursor, search },
  } = options;

  const nerrow = match(options)
    .with(
      { ctx: { payload: P._ } },
      ({
        ctx: {
          payload: {
            profile: { id: profile_id },
          },
        },
        input: { filter },
      }) => profile_filter(profile_id, filter),
    )
    .otherwise(() => ({}));

  const search_where: ExchangeWhere = {
    OR: [
      { title: { contains: search ?? "", mode: "insensitive" } },
      {
        description: { contains: search ?? "", mode: "insensitive" },
      },
      { location: { contains: search ?? "", mode: "insensitive" } },
      {
        owner: {
          profile: {
            name: { contains: search ?? "", mode: "insensitive" },
          },
        },
      },
    ],
  };

  const active_exchanges_where: ExchangeWhere = {
    OR: [{ expiry_date: null }, { expiry_date: { gte: endOfYesterday() } }],
  };
  const category_where: ExchangeWhere = category
    ? { category: { slug: category } }
    : {};

  const items = await prisma.exchange.findMany({
    ...(cursor ? { cursor: { id: cursor } } : {}),
    orderBy: [{ expiry_date: "asc" }, { created_at: "desc" }],
    take: limit + 1,
    where: {
      AND: [search_where, active_exchanges_where, category_where, nerrow],
    },
  });

  let nextCursor: typeof cursor | undefined = undefined;
  if (items.length > limit) {
    const nextItem = items.pop()!;
    nextCursor = nextItem.id;
  }

  return { data: items, nextCursor };
}

function profile_filter(
  profile_id: string,
  filter: z.infer<typeof Exchange_Filter>,
) {
  return match(filter)
    .with(Exchange_Filter.Enum.ALL, (): Prisma.ExchangeWhereInput => ({}))
    .with(
      Exchange_Filter.Enum.DATE_FLEXIBLE,
      (): Prisma.ExchangeWhereInput => ({ expiry_date: null }),
    )
    .with(
      Exchange_Filter.Enum.DATE_LIMITED,
      (): Prisma.ExchangeWhereInput => ({ expiry_date: { not: null } }),
    )
    .with(
      Exchange_Filter.Enum.MY_CIRCLES,
      (): Prisma.ExchangeWhereInput => ({
        owner: { profile: { followed_by: { some: { id: profile_id } } } },
      }),
    )
    .with(
      Exchange_Filter.Enum.ONLINE,
      (): Prisma.ExchangeWhereInput => ({ is_online: true }),
    )
    .with(
      Exchange_Filter.Enum.ON_SITE,
      (): Prisma.ExchangeWhereInput => ({ is_online: false }),
    )
    .with(
      Exchange_Filter.Enum.WITH_RETURN,
      (): Prisma.ExchangeWhereInput => ({ return: { isNot: null } }),
    )
    .with(
      Exchange_Filter.Enum.WITHOUT_RETURN,
      (): Prisma.ExchangeWhereInput => ({ return: null }),
    )
    .exhaustive();
}
