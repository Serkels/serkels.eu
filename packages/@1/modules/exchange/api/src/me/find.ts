//

import { next_auth_procedure } from "@1.modules/trpc";
import type { Prisma } from "@prisma/client";
import { z } from "zod";

//

export const find = next_auth_procedure
  .input(
    z.object({
      cursor: z.string().optional(),
      limit: z.number().min(1).max(10).default(10),
      search: z.string().optional(),
    }),
  )
  .query(async ({ input, ctx: { payload, prisma } }) => {
    const { profile } = payload;
    const { cursor, limit, search } = input;

    const { id: studient_id } = await prisma.studient.findUniqueOrThrow({
      select: { id: true },
      where: { profile_id: profile.id },
    });

    const search_where: Prisma.ExchangeWhereInput = search
      ? {
          OR: [
            { title: { contains: search, mode: "insensitive" } },
            { description: { contains: search, mode: "insensitive" } },
            { location: { contains: search, mode: "insensitive" } },
            {
              owner: {
                profile: { name: { contains: search, mode: "insensitive" } },
              },
            },
          ],
        }
      : {};

    const deal_releated_to_me_where: Prisma.DealWhereInput = {
      OR: [
        { participant_id: studient_id },
        {
          parent: { owner_id: studient_id },
        },
      ],
    };

    const find_cursor = cursor
      ? ({
          participant_per_exchange: {
            parent_id: cursor,
            participant_id: profile.id,
          },
        } as Prisma.DealWhereUniqueInput)
      : {};

    const deals = await prisma.deal.findMany({
      ...find_cursor,
      orderBy: { updated_at: "desc" },
      select: { parent: true },
      take: limit + 1,
      where: { ...deal_releated_to_me_where, parent: { ...search_where } },
      distinct: ["parent_id"],
    });

    const data = deals.map(({ parent }) => parent);

    let next_cursor: typeof cursor | undefined = undefined;
    if (data.length > limit) {
      const next_item = data.pop()!;
      next_cursor = next_item.id;
    }

    return { data, next_cursor };
  });
