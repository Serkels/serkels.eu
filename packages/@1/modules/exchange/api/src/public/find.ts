//

import type { Prisma } from "@1.infra/database";
import { Exchange_Filter } from "@1.modules/exchange.domain";
import {
  maybe_session_procedure,
  router,
  with_next_cursor,
} from "@1.modules/trpc";
import { endOfYesterday } from "date-fns";
import { match, P } from "ts-pattern";
import { z } from "zod";

//

const input_schema = z.object({
  category: z.string().optional(),
  cursor: z.string().optional(),
  filter: Exchange_Filter.default(Exchange_Filter.Enum.ALL),
  limit: z.number().min(1).max(10).default(10),
  search: z.string().optional(),
});

//

export default router({
  find: maybe_session_procedure
    .input(input_schema)
    .query(async ({ ctx: { session, prisma }, input }) => {
      const { limit, category, cursor, search } = input;
      const nerrow = match({ session, input })
        .with(
          { session: { profile: { role: "STUDENT" } } },
          ({
            session: {
              profile: { id: profile_id },
            },
            input: { filter },
          }) => profile_filter(profile_id, filter),
        )
        .otherwise(() => ({}));

      const search_where: Prisma.ExchangeWhereInput = {
        OR: [
          { title: { contains: search ?? "", mode: "insensitive" } },
          {
            description: { contains: search ?? "", mode: "insensitive" },
          },
          { location: { contains: search ?? "", mode: "insensitive" } },
          {
            owner: {
              profile: {
                name: { contains: search ?? "", mode: "insensitive" },
              },
            },
          },
        ],
      };

      const active_exchanges_where: Prisma.ExchangeWhereInput = {
        is_active: true,
        OR: [{ expiry_date: null }, { expiry_date: { gte: endOfYesterday() } }],
      };
      const category_where: Prisma.ExchangeWhereInput = category
        ? { category: { slug: category } }
        : {};

      const where_not_in_profile_blacklist = match(session?.profile)
        .with(
          { id: P.select() },
          (profile_id): Prisma.ExchangeWhereInput => ({
            AND: [
              { owner: { profile: { blacklist: { none: { profile_id } } } } },
              {
                owner: {
                  profile: {
                    blacklisted_by: { none: { owner_id: profile_id } },
                  },
                },
              },
            ],
          }),
        )
        .otherwise(() => ({}));

      const items = await prisma.exchange.findMany({
        ...(cursor ? { cursor: { id: cursor } } : {}),
        orderBy: [
          { expiry_date: { sort: "asc", nulls: "last" } },
          { created_at: "desc" },
          { id: "desc" },
        ],
        take: limit + 1,
        where: {
          AND: [
            where_not_in_profile_blacklist,
            search_where,
            active_exchanges_where,
            category_where,
            nerrow,
          ],
        },
      });

      return with_next_cursor(limit, items)((item) => item.id);
    }),
});

function profile_filter(
  profile_id: string,
  filter: z.infer<typeof Exchange_Filter>,
) {
  return match(filter)
    .with(Exchange_Filter.Enum.ALL, (): Prisma.ExchangeWhereInput => ({}))
    .with(
      Exchange_Filter.Enum.DATE_FLEXIBLE,
      (): Prisma.ExchangeWhereInput => ({ expiry_date: null }),
    )
    .with(
      Exchange_Filter.Enum.DATE_LIMITED,
      (): Prisma.ExchangeWhereInput => ({ expiry_date: { not: null } }),
    )
    .with(
      Exchange_Filter.Enum.MY_CIRCLES,
      (): Prisma.ExchangeWhereInput => ({
        owner: { profile: { in_contact_with: { some: { id: profile_id } } } },
      }),
    )
    .with(
      Exchange_Filter.Enum.ONLINE,
      (): Prisma.ExchangeWhereInput => ({ is_online: true }),
    )
    .with(
      Exchange_Filter.Enum.ON_SITE,
      (): Prisma.ExchangeWhereInput => ({ is_online: false }),
    )
    .with(
      Exchange_Filter.Enum.WITH_RETURN,
      (): Prisma.ExchangeWhereInput => ({ return: { isNot: null } }),
    )
    .with(
      Exchange_Filter.Enum.WITHOUT_RETURN,
      (): Prisma.ExchangeWhereInput => ({ return: null }),
    )
    .exhaustive();
}
