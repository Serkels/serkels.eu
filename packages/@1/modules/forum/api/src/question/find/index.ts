//

import type { Prisma } from "@1.infra/database";
import { Forum_Filter } from "@1.modules/forum.domain";
import { procedure } from "@1.modules/trpc";
import { match } from "ts-pattern";
import { z } from "zod";

//

export default procedure
  .input(
    z.object({
      category: z.string().optional(),
      cursor: z.string().optional(),
      filter: Forum_Filter.default(Forum_Filter.Enum.ALL),
      limit: z.number().min(1).max(10).default(10),
      search: z.string().optional(),
      profile_id: z.string().nullable().optional(),
    }),
  )
  .query(async ({ input, ctx: { prisma } }) => {
    const { category, cursor, limit, search, filter, profile_id } = input;

    const narrow = match(filter)
      .with(Forum_Filter.Enum.ALL, (): Prisma.QuestionWhereInput => ({}))
      .with(
        Forum_Filter.Enum.APPROVED,
        (): Prisma.QuestionWhereInput => ({
          NOT: { accepted_answer_id: null },
        }),
      )
      .with(
        Forum_Filter.Enum.AWNSERED,
        (): Prisma.QuestionWhereInput => ({
          NOT: [{ answers: { none: {} } }],
        }),
      )
      .with(
        Forum_Filter.Enum.LAST_QUESTIONS,
        (): Prisma.QuestionWhereInput => ({}),
      )
      .with(
        Forum_Filter.Enum.MINE,
        (): Prisma.QuestionWhereInput =>
          profile_id
            ? {
                owner: { profile_id },
              }
            : {},
      )
      .with(
        Forum_Filter.Enum.NOT_APPROVED,
        (): Prisma.QuestionWhereInput => ({
          accepted_answer_id: null,
        }),
      )
      .exhaustive();

    const items = await prisma.question.findMany({
      ...(cursor ? { cursor: { id: cursor } } : {}),
      orderBy: { created_at: "desc" },
      take: limit + 1,
      where: {
        OR: [
          {
            owner: {
              profile: {
                name: { contains: search ?? "", mode: "insensitive" },
              },
            },
          },
          { title: { contains: search ?? "", mode: "insensitive" } },
        ],
        ...(category ? { category: { slug: category } } : {}),
        ...narrow,
      },
    });

    let nextCursor: typeof cursor | undefined = undefined;
    if (items.length > limit) {
      const next_item = items.pop()!;
      nextCursor = next_item.id;
    }

    return { data: items, nextCursor };
  });
