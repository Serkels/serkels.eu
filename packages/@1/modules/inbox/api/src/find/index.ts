// //

import type { Prisma, PrismaClient } from "@1.infra/database";
import { next_auth_procedure, with_next_cursor } from "@1.modules/trpc";
import { match, P } from "ts-pattern";
import { z } from "zod";

// //

const get_student_id = (prisma: PrismaClient) => (profile_id: string) => {
  return prisma.student.findFirstOrThrow({
    select: { id: true },
    where: { profile_id },
  });
};

// const get_inbox_threads = (prisma: PrismaClient) => (student_id: string) => {
//   return prisma.inboxThread.findMany({
//     where: { owner_id: student_id },
//   });
// };

// //

const input_schema = z.object({
  cursor: z.string().optional(),
  limit: z.number().min(1).max(10).default(10),
  search: z.string().optional(),
});

const where_search_input = (search: string | undefined) =>
  match(search)
    .with(
      P.string,
      (value): Prisma.InboxThreadWhereInput => ({
        OR: [
          {
            thread: {
              participants: {
                some: { name: { contains: value, mode: "insensitive" } },
              },
            },
          },
          {
            thread: {
              messages: {
                some: {
                  content: { contains: value, mode: "insensitive" },
                },
              },
            },
          },
        ],
      }),
    )
    .otherwise(() => ({}));

export default next_auth_procedure
  .input(input_schema)
  .query(async ({ ctx: { payload, prisma }, input }) => {
    const { profile } = payload;
    const { cursor, limit, search } = input;
    const { id: student_id } = await get_student_id(prisma)(profile.id);

    const search_where = where_search_input(search);

    const data = await prisma.inboxThread.findMany({
      ...(cursor
        ? {
            cursor: {
              owner_id_thread_id: { owner_id: student_id, thread_id: cursor },
            },
          }
        : {}),
      orderBy: [{ thread: { updated_at: "desc" } }],
      include: { thread: { select: { id: true, updated_at: true } } },
      take: limit + 1,
      where: { owner_id: student_id, ...search_where },
    });

    return with_next_cursor(limit, data)((item) => item.thread_id);
  });
