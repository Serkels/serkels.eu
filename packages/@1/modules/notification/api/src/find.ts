//

import { next_auth_procedure } from "@1.modules/trpc";
import { z } from "zod";

//

export default next_auth_procedure
  .input(
    z.object({
      cursor: z.string().optional(),
      limit: z.number().min(1).max(10).default(10),
    }),
  )
  .query(async ({ input, ctx: { prisma, payload } }) => {
    const { cursor, limit } = input;
    const { id: owner_id } = payload.profile;
    const items = await prisma.notification.findMany({
      ...(cursor ? { cursor: { id: cursor } } : {}),
      orderBy: { created_at: "desc" },
      take: limit + 1,
      where: { owner_id },
      include: {
        inbox_message: {
          select: {
            message: {
              select: {
                author: { select: { name: true } },
                thread_id: true,
              },
            },
          },
        },
      },
    });

    let prevCursor: typeof cursor | undefined = undefined;
    if (items.length > limit) {
      const nextItem = items.pop()!;
      prevCursor = nextItem.id;
    }

    return { data: items, prevCursor };
  });
