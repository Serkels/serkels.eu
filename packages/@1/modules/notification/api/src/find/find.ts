//

import type { Prisma } from "@1.infra/database";
import {
  router,
  session_procedure,
  with_previous_cursor,
} from "@1.modules/trpc";
import { z } from "zod";

//

export default router({
  find: session_procedure
    .input(
      z.object({
        cursor: z.string().optional(),
        limit: z.number().min(1).max(10).default(10),
      }),
    )
    .query(async ({ input, ctx: { prisma, session } }) => {
      const { cursor, limit } = input;
      const { id: owner_id } = session.profile;
      const message_select: Prisma.MessageSelect = {
        author: { select: { name: true, id: true, image: true } },
        thread_id: true,
      };

      const items = await prisma.notification.findMany({
        ...(cursor ? { cursor: { id: cursor } } : {}),
        orderBy: { created_at: "desc" },
        take: limit + 1,
        where: { owner_id },
        include: {
          inbox_message: { select: { message: { select: message_select } } },
          exchange_message: {
            select: {
              exchange: {
                select: {
                  id: true,
                  owner: {
                    select: {
                      profile: {
                        select: { id: true, name: true, image: true },
                      },
                    },
                  },
                  title: true,
                },
              },
              message: { select: message_select },
            },
          },
          forum_message: {
            select: {
              answer: {
                select: {
                  owner: {
                    select: {
                      profile_id: true,
                      profile: { select: { name: true, image: true } },
                    },
                  },
                  parent_id: true,
                  parent: { select: { id: true, title: true } },
                },
              },
              answer_id: true,
            },
          },
          profile_added: {
            select: {
              profile: { select: { name: true, id: true, image: true } },
            },
          },
        },
      });

      return with_previous_cursor(limit, items)((item) => item.id);
    }),
});
