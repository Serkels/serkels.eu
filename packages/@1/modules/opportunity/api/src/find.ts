//

import { Partner_Filter } from "@1.modules/opportunity.domain";
import {
  next_auth_procedure,
  procedure,
  router,
  type inferProcedureInput,
} from "@1.modules/trpc";
import { startOfToday } from "date-fns/startOfToday";
import { P, match } from "ts-pattern";
import { z } from "zod";

//

const public_schema = z.object({
  category: z.string().optional(),
  cursor: z.string().optional(),
  limit: z.number().min(1).max(12).default(12),
  search: z.string().optional(),
});

const public_procedure = procedure.input(public_schema);

const private_procedure = next_auth_procedure.input(
  public_schema.extend({
    filter: Partner_Filter.default(Partner_Filter.Enum.ALL),
  }),
);

export default router({
  public: public_procedure.query(find_resolver),
  private: private_procedure.query(find_resolver),
});

//

async function find_resolver(
  options:
    | inferProcedureInput<typeof public_procedure>
    | inferProcedureInput<typeof private_procedure>,
) {
  const {
    ctx: { prisma },
    input: { limit, category, cursor, search },
  } = options;

  type OpportunityWhereInput = NonNullable<
    Parameters<typeof prisma.opportunity.findMany>[0]
  >["where"];

  const nerrow = match(options)
    .with(
      { ctx: { payload: P._ } },
      ({
        ctx: {
          payload: {
            profile: { id: profile_id },
          },
        },
        input: { filter },
      }) =>
        match(filter)
          .with(
            Partner_Filter.Enum.MY_OPPORTUNITIES,
            (): OpportunityWhereInput => ({ owner: { profile_id } }),
          )
          .with(Partner_Filter.Enum.ALL, () => ({}))
          .otherwise(() => ({})),
    )
    .otherwise(() => ({}));

  const items = await prisma.opportunity.findMany({
    ...(cursor ? { cursor: { id: cursor } } : {}),
    orderBy: { expiry_date: "asc" },
    take: limit + 1,
    where: {
      OR: [
        { title: { contains: search ?? "", mode: "insensitive" } },
        { description: { contains: search ?? "", mode: "insensitive" } },
      ],
      expiry_date: { gte: startOfToday() },
      ...(category ? { category: { slug: category } } : {}),
      ...nerrow,
    },
    include: {
      category: true,
      owner: { include: { profile: true } },
    },
  });
  let nextCursor: typeof cursor | undefined = undefined;
  if (items.length > limit) {
    const nextItem = items.pop()!;
    nextCursor = nextItem.id;
  }
  return { data: items, nextCursor };
}
