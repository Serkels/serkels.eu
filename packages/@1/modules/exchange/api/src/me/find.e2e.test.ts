//

import {
  create_douglas_student,
  create_film_category,
  create_joedart_student,
  create_masterclass_category,
  create_slap_exchange,
  create_the_creator_exchange,
} from "@1.infra/database/seeding";
import prisma, { empty_database, migrate } from "@1.infra/database/testing";
import {
  create_nextauth_header,
  createCallerFactory,
  router,
} from "@1.modules/trpc";
import {
  afterEach,
  beforeAll,
  beforeEach,
  expect,
  setSystemTime,
  test,
} from "bun:test";
import { find } from "./find";

//

const NEXTAUTH_SECRET = "ðŸ”‘";

beforeAll(() => {
  process.env["NEXTAUTH_SECRET"] = NEXTAUTH_SECRET;
});

//

beforeAll(empty_database);
beforeAll(migrate);
beforeAll(() => {
  setSystemTime(new Date("2011-11-11"));
});

//

let nextauth_header: Awaited<ReturnType<typeof create_nextauth_header>>;

beforeAll(async () => {
  nextauth_header = await create_nextauth_header({
    secret: "ðŸ”‘",
    token: {
      from: import.meta.url,
      profile: { id: "douglas_profile_id", role: "STUDENT" },
    },
  });
});

//

test("return my latest exchanges", async () => {
  const caller = createCallerFactory(router({ find }));
  const trpc = caller({
    headers: { ...nextauth_header },
    prisma,
  } as any);
  const response = await trpc.find({});
  expect(response).toMatchSnapshot();
});

test("return a valid cursor", async () => {
  const caller = createCallerFactory(router({ find }));
  const trpc = caller({
    headers: { ...nextauth_header },
    prisma,
  } as any);
  const first_response = await trpc.find({ limit: 1 });
  expect(first_response).toMatchSnapshot("first_response");

  const second_response = await trpc.find({
    limit: 1,
    cursor: first_response.next_cursor,
  });
  expect(second_response).toMatchSnapshot("second_response");
});

test("should not return archived exchanges", async () => {
  await prisma.exchangeThread.updateMany({
    data: { is_archived: true },
    where: {
      id: {
        in: [
          "slap_exchange_id_douglas_student_id_exchange_thread_id",
          "the_creator_exchange_id_douglas_student_id_exchange_thread_id",
        ],
      },
    },
  });
  const caller = createCallerFactory(router({ find }));
  const trpc = caller({
    headers: { ...nextauth_header },
    prisma,
  } as any);
  const exchange = await trpc.find({});
  expect(exchange).toMatchSnapshot();
});

//

beforeEach(async function seed() {
  await create_masterclass_category(prisma);
  await create_film_category(prisma);
  const douglas_student_id = await create_douglas_student(prisma);
  const joedart_student_id = await create_joedart_student(prisma);
  const { id: slap_exchange_id } = await create_slap_exchange(
    prisma,
    joedart_student_id,
  );
  const { id: the_creator_exchange_id } = await create_the_creator_exchange(
    prisma,
    douglas_student_id,
  );

  await prisma.exchangeThread.create({
    data: {
      id: "slap_exchange_id_douglas_student_id_exchange_thread_id",
      deal: {
        create: {
          id: "slap_exchange_id_douglas_student_id_deal_id",
          parent: { connect: { id: slap_exchange_id } },
          participant: { connect: { id: douglas_student_id } },
        },
      },
      owner: { connect: { id: douglas_student_id } },
      thread: {
        create: {
          id: "slap_exchange_id_douglas_student_id_thread_id",
          updated_at: new Date("2011-11-11"),
        },
      },
    },
  });
  await prisma.exchangeThread.create({
    data: {
      id: "the_creator_exchange_id_douglas_student_id_exchange_thread_id",
      deal: {
        create: {
          id: "the_creator_exchange_id_douglas_student_id_deal_id",
          parent: { connect: { id: the_creator_exchange_id } },
          participant: { connect: { id: douglas_student_id } },
        },
      },
      owner: { connect: { id: douglas_student_id } },
      thread: {
        create: {
          id: "the_creator_exchange_id_douglas_student_id_thread_id",
          updated_at: new Date("2011-11-10"),
        },
      },
    },
  });
});

afterEach(async function empty_delete_entries() {
  await prisma.category.deleteMany({});
  await prisma.deal.deleteMany({});
  await prisma.exchange.deleteMany({});
  await prisma.exchangeThread.deleteMany({});
  await prisma.thread.deleteMany({});
  await prisma.user.deleteMany({});
});
