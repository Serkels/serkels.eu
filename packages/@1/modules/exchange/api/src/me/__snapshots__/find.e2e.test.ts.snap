// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`return my latest exchanges 1`] = `
{
  "data": [
    {
      "exchange": {
        "category_id": "masterclass_category_id",
        "created_at": 2011-11-11T00:00:00.000Z,
        "description": "https://vulf.co/courses/dart/lectures/45940163",
        "expiry_date": null,
        "id": "slap_exchange_id",
        "is_active": true,
        "is_online": true,
        "location": null,
        "owner_id": "joedart_student_id",
        "places": 1,
        "return_id": null,
        "title": "Slap",
        "type": "PROPOSAL",
        "updated_at": 2011-11-11T00:00:00.000Z,
      },
      "is_unread": false,
      "last_thread_update": 2011-11-11T00:00:00.000Z,
    },
  ],
  "next_cursor": undefined,
}
`;

exports[`should not return archived exchanges 1`] = `
{
  "data": [],
  "next_cursor": undefined,
}
`;
